packages/language-support/src/autocompletion/completionCoreCompletions.ts

```diff
@@ -312,10 +296,16 @@ const parameterCompletions = (
     .filter(([_, paramType]) =>
       isExpectedParameterType(expectedType, paramType),
     )
-    .map(([paramName]) => ({
-      label: `$${paramName}`,
-      kind: CompletionItemKind.Variable,
-    }));
+    .map(([paramName]) => {
+      const backtickedName = backtickIfNeeded(paramName);
+      return {
+        label: `$${paramName}`,
+        kind: CompletionItemKind.Variable,
+        ...(backtickedName != undefined
```

Comment:
This comparison needs to be `!==` I think.

You can just do `backtickedName ? { insertText: `$${backtickIfNeeded(paramName)}` } : {}`